src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs

```diff
@@ -219,6 +219,14 @@ private void MethodChecks(MethodDeclarationSyntax syntax, Binder withTypeParamsB
                     var loc = parameterSyntax.Type.Location;
                     diagnostics.Add(ErrorCode.ERR_BadTypeforThis, loc, parameter0Type);
                 }
+                else if (this.Parameters[0].RefKind == RefKind.Ref && !parameter0Type.IsValueType)
+                {
+                    diagnostics.Add(ErrorCode.ERR_RefExtensionMustBeValueTypeOrConstrainedToOne, location, Name);
+                }
+                else if (this.Parameters[0].RefKind == RefKind.RefReadOnly && parameter0Type.TypeKind != TypeKind.Struct)
```

Comment:
>else if (this.Parameters[0].RefKind == RefKind.RefReadOnly && parameter0Type.TypeKind != TypeKind.Struct) [](start = 16, length = 105)

Are we testing consumption of extension methods that do not meet this requirement and the one above?